tools/goworksync.sh

```diff
@@ -1,19 +1,13 @@
 #!/bin/bash
 
-GOBIN=$1
-
-(
-	cd "$BUILD_WORKSPACE_DIRECTORY" || exit
-	GO="$(bazel info execution_root)/$GOBIN"
-
-	find . -name go.mod |
-		while IFS= read -r -d '' f; do
-			d=$(dirname "$f")
-			(
-				cd "$d" || exit
-				"$GO" mod tidy
-			)
-		done
-
-	"$GO" work sync
-)
+find . -name go.mod -print0 |
+	while IFS= read -r -d '' f; do
+		d=$(dirname "$f")
+		(
+			cd "$d" || exit
+			echo "Running 'go mod tidy' in $d"
+			bazel run @rules_go//go -- mod tidy
+		)
+	done
+
+bazel run @rules_go//go -- work sync
```

Comment:
I think it's not safe to remove the `cd "$BUILD_WORKSPACE_DIRECTORY"` here, otherwise subsequent `cd` would navigate to the wrong directories.

Keep in mind that you may be running this tool with `PWD=//<root_package>`, but you can also be running from `PWD=//some/deep/package`, thus the initial cd is needed.

Reply:
Fixed